src/main/java/com/educare/unitylend/controller/RepaymentTransactionController.java

```diff
@@ -1,74 +1,162 @@
 package com.educare.unitylend.controller;
 
 import com.educare.unitylend.Exception.ControllerException;
+import com.educare.unitylend.Exception.ServiceException;
+import com.educare.unitylend.model.LendTransaction;
 import com.educare.unitylend.model.RepaymentTransaction;
+import com.educare.unitylend.service.RepaymentTransactionService;
 import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.math.BigDecimal;
+import java.util.List;
+
 @Slf4j
 @AllArgsConstructor
 @RestController
 @RequestMapping("/repayment-transaction")
-public class RepaymentTransactionController extends BaseController{
+public class RepaymentTransactionController extends BaseController {
+    RepaymentTransactionService repaymentTransactionService;
 
     /**
-     * @return List of all the available {@link RepaymentTransaction}
-     * @throws ControllerException : Exception to be thrown from controller in case of any exception
+     * API endpoint for creating a repayment transaction.
+     *
+     * @param borrowRequestId The ID of the borrow request.
+     * @return ResponseEntity<String> Indicates the success or failure of the transaction creation.
+     * @throws ControllerException If an error occurs during the transaction creation process.
      */
+    @PostMapping("/repay")
+    public ResponseEntity<String> initiateRepaymentTransaction(
+            @RequestParam String borrowRequestId
+    ) throws ControllerException {
+        try {
+            if (borrowRequestId.isEmpty()) {
```

Comment:
There is no need to add sepcific null checks inside the function. 
You can add  `@RequestParam (isRequired=True) ` and this can be handled by spring

Make this change at all the required places.